bbc/digital-paper-edit-storybook
# Install react-dom & add babel plugin

**Is your Pull Request request related to another [issue](https://github.com/bbc/digital-paper-edit-storybook/issues) in this repository ?**      
https://github.com/bbc/digital-paper-edit-stt-proxy/issues/40

**Describe what the PR does**
Add missing react-dom dependency.
Adds all react-related dependencies to peerDependencies.
When `npm run link:setup` is run, the react-related dependencies are removed, so there's no clash of React versions when running the firebase app.
Copies package.json to the dist directory when linked or published and removes react-related dependencies.

**State whether the PR is ready for review or whether it needs extra work**    
RFR

**Steps to test**
Ensure storybook and firebase projects are both using Node 12.18.4 (`nvm use 12.18.4`).

1. In the storybook repo:
```
rm -rf node_modules && npm i && npm start
```
2. Check the storybook app works as expected. Then Ctrl-C out and run:
```
npm run link:setup
```
Check that ./dist exists, has .js and .js.map files and a package.json without react dependencies (react, react-dom, react-router & react-router dom)

3. In another terminal window, still in the project root, run:
```
npm link
```
4. In the firebase repo, start the app:
```
npm run dev:react
```
Check the app works as expected.

5. To test changes to storybook are reflected in firebase, in the storybook repo, make a change or add a console.log in one of the components, then check that shows in the firebase app.

6. When finished, from the storybook repo, run:
```
npm run unlink
```
This will undo the npm link and install the react deps back into the project.
